Source/WebKit/GPUProcess/webrtc/RemoteMediaRecorder.cpp

```diff
@@ -37,7 +37,14 @@
 #include <WebCore/WebAudioBufferList.h>
 #include <wtf/CompletionHandler.h>
 
-#define MESSAGE_CHECK(assertion) MESSAGE_CHECK_BASE(assertion, (connection ? &connection->connection() : nullptr))
+#define MESSAGE_CHECK(assertion) do { \
+    if (auto didFailAssertion = !(assertion)) { \
+        auto connection = m_gpuConnectionToWebProcess.get(); \
+        if (!connection) \
+            return; \
+        MESSAGE_CHECK_BASE(!didFailAssertion, &connection->connection()); \
```

Comment:
Maybe we can add null check for connection in `MESSAGE_CHECK_BASE` instead? It seems more robust than adding the null check for every `MESSAGE_CHECK`

Reply:
I don't think we can do that because MESSAGE_CHECK_BASE takes Connection as an argument, not GPUConnectionToWebProcess.

Reply:
I mean something like:
```
#define MESSAGE_CHECK(assertion) MESSAGE_CHECK_BASE(assertion, connection())
IPC::Connection* RemoteMediaRecorder::connection() {
    return m_gpuConnectionToWebProcess ? &m_gpuConnectionToWebProcess->connection() : nullptr;
}
void RemoteMediaRecorder::audioSamplesAvailable(MediaTime time, uint64_t numberOfFrames)
{
    MESSAGE_CHECK(m_ringBuffer);
    ....
}
...
#define MESSAGE_CHECK_COMPLETION_BASE(assertion, connection, completion) do { \
    if (UNLIKELY(!(assertion))) { \
        RELEASE_LOG_FAULT(IPC, __FILE__ " " CONNECTION_STRINGIFY_MACRO(__LINE__) ": Invalid message dispatched %s", WTF_PRETTY_FUNCTION); \
        if (connection)
            connection->markCurrentlyDispatchedMessageAsInvalid(); \
        { completion; } \
        return; \
    } \
} while (0)
```

Reply:
No, that would be racy since `m_gpuConnectionToWebProcess` is a `ThreadSafeWeakPtr`.

Reply:
I'd remove the `connection` local variable from `RemoteMediaRecorder::audioSamplesStorageChanged` as we discussed in person.